ddtrace/contrib/pytest/_plugin_v1.py

```diff
@@ -0,0 +1,861 @@
+            _mark_not_skipped(_get_parent(item))
+            span.set_tag_str(test.STATUS, test.Status.FAIL.value)
+            if xfail and not has_skip_keyword and not item.config.option.runxfail:
+                # XPass (strict=True) are recorded failed by pytest, longrepr contains reason
+                span.set_tag_str(XFAIL_REASON, getattr(result, "longrepr", "XFail"))
+                span.set_tag_str(test.RESULT, test.Status.XPASS.value)
+            if call.excinfo:
+                span.set_exc_info(call.excinfo.type, call.excinfo.value, call.excinfo.tb)
+
+    @staticmethod
+    @pytest.hookimpl(trylast=True)
+    def pytest_ddtrace_get_item_module_name(item):
+        pytest_module_item = _find_pytest_item(item, pytest.Module)
+        pytest_package_item = _find_pytest_item(pytest_module_item, pytest.Package)
+
+        if _module_is_package(pytest_package_item, pytest_module_item):
+            if _is_pytest_8_or_later():
+                # pytest 8.0.0 no longer treats Packages as Module/File, so we replicate legacy behavior by
+                # concatenating parent package names in reverse until we hit a non-Package-type item
+                # https://github.com/pytest-dev/pytest/issues/11137
+                package_names = []
+                current_package = pytest_package_item
+                while isinstance(current_package, pytest.Package):
+                    package_names.append(str(current_package.name))
+                    current_package = current_package.parent
+
+                return ".".join(package_names[::-1])
+
+            return pytest_package_item.module.__name__
+
+        return pytest_module_item.nodeid.rpartition("/")[0].replace("/", ".")
+
+    @staticmethod
+    @pytest.hookimpl(trylast=True)
+    def pytest_ddtrace_get_item_suite_name(item):
+        """
+        Extract suite name from a `pytest.Item` instance.
+        If the module path doesn't exist, the suite path will be reported in full.
+        """
+        pytest_module_item = _find_pytest_item(item, pytest.Module)
+        test_module_path = _get_module_path(pytest_module_item)
+        if test_module_path:
+            if not pytest_module_item.nodeid.startswith(test_module_path):
+                log.warning(
+                    "Suite path is not under module path: '%s' '%s'", pytest_module_item.nodeid, test_module_path
+                )
+            return get_relative_or_absolute_path_for_path(pytest_module_item.nodeid, test_module_path)
+        return pytest_module_item.nodeid
+
+    @staticmethod
+    @pytest.hookimpl(trylast=True)
+    def pytest_ddtrace_get_item_test_name(item):
+        """Extract name from item, prepending class if desired"""
+        if hasattr(item, "cls") and item.cls:
+            if item.config.getoption("ddtrace-include-class-name") or item.config.getini("ddtrace-include-class-name"):
+                return "%s.%s" % (item.cls.__name__, item.name)
+        return item.name
```

Comment:
:question: is this file mostly what we had in `ddtrace/contrib/pytest/plugin.py`? Or do we have any other change?

Reply:
It's almost exactly the same thing.

Reply:
The practical difference is that this defines the plugin as a class, and moves the module-level plugin hook functions to be static methods on the class, but aside from that, there should be no functionality difference.